### PR TITLE
create checkAuth controllers

### PR DESCRIPTION
**Pull request by farah2003**
relates #11 
- [x] create checkAuth controllers

- [x] test route
### Response:
{ 
data :{
     user
    }}

### CODE REVIEW

File: server/controllers/checkAuth.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const checkAuth = (req:any, res:Response) => {
+  res.status(200).json({
+    data: [{
+      id: req.user.id,
+      name: req.user.name,
+      isAdmin: req.user.isAdmin,
+    }],
+  });
+};
```
Comment (Raghad-Mezied) [inline server/controllers/checkAuth.ts:11]: You don't need to send the status code of 200 as it is the default and you already have the req.user object why you are creating a new one

File: server/tests/route.test.ts [hunk 2]
```diff
@@ -110,6 +110,22 @@ describe('POST/login', () => {
     );
   });
 });
+describe('GET/checkAuth', () => {
+  test('Authorized', async () => {
+    const response = await request(app)
+      .get('/api/checkAuth')
+      .set('Cookie', [' ACCESS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NiwibmFtZSI6InNhcmFhYWFhIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTY1MTA4ODczNCwiZXhwIjoxNjUzNjgwNzM0fQ.0WLTAq9cljwNyJE2dUOnTa1rzBgNVvpMIE1GwXfNG8U'])
+      .expect(200);
+    expect(response.body.message).toBe('Authorized user');
```
Comment (Raghad-Mezied) [inline server/tests/route.test.ts:119]: Test the data too, if it is as you expect.

File: server/tests/route.test.ts [hunk 3]
```diff
@@ -110,6 +110,22 @@ describe('POST/login', () => {
...
+  });
+  test('unAuthorized uer ', async () => {
+    const response = await request(app)
+      .get('/api/checkAuth')
+      .set('Cookie', [' ACCESS_TOKEN=66666eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NiwibmFtZSI6InNhcmFhYWFhIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTY1MTA4ODczNCwiZXhwIjoxNjUzNjgwNzM0fQ.0WLTAq9cljwNyJE2dUOnTa1rzBgNVvpMIE1GwXfNG8U'])
+      .expect(200);
+    expect(response.body.message).toBe('Authorized user');
+  });
+});
```
Comment (Raghad-Mezied) [inline server/tests/route.test.ts:128]: Here you should test the unAuthorized user, but this is the same as the first one

File: server/controllers/auth/checkAuth.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { Response } from 'express';
+
+const checkUser = (req:any, res:Response) => {
+  res.json({ data: req.user, message: 'Authorized user' });
```
Comment (Raghad-Mezied) [inline server/controllers/auth/checkAuth.ts:4]: Destructure the user and you don't need to send a message here.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/tests/route.test.ts [hunk 5]
```diff
@@ -110,6 +110,23 @@ describe('POST/login', () => {
...
+      exp: 1653680734, iat: 1651088734, id: 6, isAdmin: false, name: 'saraaaaa',
+    });
+  });
+  test('unAuthorized uer ', async () => {
+    await request(app)
+      .get('/api/checkAuth')
+      .set('Cookie', [' ACCESS_TOKEN=66666eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NiwibmFtZSI6InNhcmFhYWFhIiwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTY1MTA4ODczNCwiZXhwIjoxNjUzNjgwNzM0fQ.0WLTAq9cljwNyJE2dUOnTa1rzBgNVvpMIE1GwXfNG8U'])
+      .expect(401);
+  });
+});
```
Comment (Raghad-Mezied) [inline server/tests/route.test.ts:129]: Check the case where there is no `ACCESS_TOKEN` at all

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]
Good job farah 👏

**Review by Raghad-Mezied — APPROVED** [review]

**Review by HaniOlwan — APPROVED** [review]

**Review by DeenaAlghazali — APPROVED** [review]

**Pull request merged**